src/main/java/org/veupathdb/service/sr/reference/ReferenceSequenceSpec.java

```diff
@@ -14,33 +16,23 @@ public class ReferenceSequenceSpec {
   private final long maxTotalBasesPerRequest;
   private final boolean isStranded;
 
-  public ReferenceSequenceSpec(String name, long maxSequencesPerRequest, long maxTotalBasesPerRequest, boolean isStranded){
+  public ReferenceSequenceSpec(String name, long maxSequencesPerRequest, long maxTotalBasesPerRequest, boolean isStranded) {
     this.name = name;
     this.maxSequencesPerRequest = maxSequencesPerRequest;
     this.maxTotalBasesPerRequest = maxTotalBasesPerRequest;
     this.isStranded = isStranded;
   }
 
-  public <T extends BEDFeature> void validateFeatures(Collection<T> features){
-    if(! isStranded && features.stream().anyMatch(f -> ! f.getStrand().equals(Strand.NONE))){
-      throw new BadRequestException("Requested stranded feature on an unstranded reference sequence " + name );
+  public <T extends BEDFeature> void validateFeatures(Collection<T> features) {
+    if (!isStranded && features.stream().anyMatch(f -> !f.getStrand().equals(Strand.NONE))) {
+      throw new BadRequestException("Requested stranded feature on an unstranded reference sequence " + name);
     }
-    if(features.size() > maxSequencesPerRequest){
+    if (features.size() > maxSequencesPerRequest) {
       throw new BadRequestException("Requested " + features.size() + " features but reference sequence " + name + " has a per-request limit of " + maxSequencesPerRequest);
     }
-    var totalRequestedBases = features.stream().mapToInt(f -> f.getLengthOnReference()).sum();
-    if (totalRequestedBases > maxTotalBasesPerRequest){
-      throw new BadRequestException("Requested " + totalRequestedBases + " total bases but reference sequence " + name + " has a per-request limit of " + maxTotalBasesPerRequest);
-    }
-
-    Optional<T> featureWithEndBeforeStart = features.stream()
-        .filter(feature -> feature.getEnd() < feature.getStart())
-        .findFirst();
-    if (featureWithEndBeforeStart.isPresent()) {
-      throw new BadRequestException(String.format("Malformed feature; start point %d lies after end point %d",
-          featureWithEndBeforeStart.get().getStart(),
-          featureWithEndBeforeStart.get().getEnd()));
+    var totalRequestedBases = features.stream().mapToInt(Locatable::getLengthOnReference).sum();
+    if (totalRequestedBases > maxTotalBasesPerRequest) {
+      throw new BadRequestException("You requested a sequence of length " + totalRequestedBases + ". The limit for " + name + " is " + maxTotalBasesPerRequest + " each.");
```

Comment:
I feel like the old message was clearer here.  I prefer not to use "you" in error messages; passive voice is more friendly and often clearer.  In this case, "You requested a sequence of length 12.  The limit for sequenceName is 10 each."  10 what?  On each what?  Feel like we should use the verbiage in the var names, like "Requested sequence length (12) exceeds the maximum (10 bases) for a single request on sequence 'sequenceName'."

Reply:
This was largely Steve's wording. The word bases was removed since these aren't all genomic sequences. Including the sequence name was a point of contention between Steve and Cristina.

I think Cristina was worried that the sequence names we have internally would be confusing to users. I left it off because the limit we have is the same for everything, but I'm happy to add it back.